### ISSUE TITLE
performance tab on windows

### ISSUE THREAD

**Issue by madethisforprism** [PrismLauncher/PrismLauncher#2832]
### Role

i play modded minecraft on a laptop and made this github account

### Suggestion

add the performance tab to the windows version

### Benefit

to allow setting the use of discrete gpu from modpack to modpack

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

from what i can tell the "performance" tab in settings is a linux only feature, i would prefer if this were not the case and i could use it on windows, im sorry if this is incorrect as this is just what ive gathered as to why i see no performance tab and im not really sure but i really really want to be able to set specific modpacks to use the discrete vs the embedded gpu (ps i made a whole github account just for this thats how you know i want it bad)

**Comment by Trial97** [comment]
Ok any suggestion on how to implement this stuff on windows?
On linux is simple: a few environment variables before the process and you are set.
From what I know on windows you need to navigate in your GPU driver setting program and tell it to run java with GPU.
Is there a way to do that programmatically?

**Comment by madethisforprism** [comment]
ahh so theres a problem with doing it, i had feared that was the reason! no i have no knowledge of the technicalities of gpu selection for either linux or windows. linux really rocks huh? well thanks for commenting, if you know there to be no way to do it and were being rhetorical then you can close this but if you were genuine then i guess keep it open for a short while longer to see if anyone knows how to do that

**Comment by maskersss** [comment]
from what i know, selecting it in windows settings works, and i'd imagine (?) should be possible for prism to do (just a guess, no clue how)
source: https://obsproject.com/kb/minecraft-java-edition-troubleshooting

though if you have it selected one way in windows settings and another way in the GPU driver program, i have no clue which one is going to take priority (or if that's two different ways to change the same setting)

**Comment by Trial97** [comment]
Yeah but is there a way to do this in code? or just a small tutorial for users to do manually?

**Comment by maskersss** [comment]
looks like these settings are stored in `HKEY_CURRENT_USER\SOFTWARE\Microsoft\DirectX\UserGpuPreferences`, at least for me, no clue if it's possible to just edit them or if you need launching prism as admin or smth

**Comment by Olivoz** [comment]
Systems with Nvidia Optimus can use the `SHIM_MCCOMPAT` environment variable set to `0x800000001`. 
0x800000000 being the override and 0x000000001 telling it to enable the GPU. You can use 0x000000000 for integrated graphics instead. For reference [this](https://github.com/CaffeineMC/sodium-fabric/blob/888d922e3f034bf97bb1c0171e8012b22a9a1be6/common/src/workarounds/java/net/caffeinemc/mods/sodium/client/compatibility/workarounds/nvidia/NvidiaWorkarounds.java#L24) is how Sodium enables the GPU.

**Comment by Trial97** [comment]
Ok that is nice can somebody who owns a dual GPU if that works for both Nvidia and AMD?
You can set that env in the instance settings:
![image](https://github.com/user-attachments/assets/060c23ef-3e1d-4d4d-91b5-e56924d66f2f)

**Comment by madethisforprism** [comment]
I TRIED IT AND IT WORKED!!!! THANK YOU ALL THANK YOU!!!!!! YOU ARE ALL SUCH MAGICAL HELPERS!!!!
even if this never gets added im still happy i have a method now thank you all for the work you do to help improve things for everyone!!!!

**Comment by madethisforprism** [comment]
oh sorry i just re-read what you were asking, i have an integrated amd gpu and a discrete nvidia gpu
